compiler/plugins/target/AMD-AIE/aie/AMDAIEObjectFifoStatefulTransform.cpp

```diff
@@ -138,26 +141,31 @@ std::optional<Value> getOptionalSharedTile(ObjectFifoLinkOp op) {
 }  // namespace
 
 class LockAnalysis {
-  DenseMap<std::pair<Value, int>, int> locksPerTile;
+ private:
+  DenseSet<std::pair<Value, int>> locks;
+  AMDAIEDeviceModel deviceModel;
+
+  void insert(LockOp l) {
+    auto id = l.getLockID().value();
+    auto tile = l.getTile();
+    locks.insert(std::pair<Value, int>(tile, id));
+  }
```

Comment:
```suggestion
  DenseSet<std::pair<Value, int8_t>> locks;
  AMDAIEDeviceModel deviceModel;

  void insert(LockOp l) {
    int8_t id = l.getLockID().value();
    Value tile = l.getTile();
    locks.insert(std::pair<Value, int8_t>(tile, id));
  }
```

Reply:
`uint8_t`

Reply:
Yeah, that needs a change in the op, but would be even better.

Reply:
oh hmm I forgot to change that - the init value is legit `int8_t` but the id should be `uint8_t`.